### PR TITLE
Move vectorizing iterator to a new automatic standard module

### PR DESCRIPTION
**Pull request by mppf**
This PR moves `vectorizeOnly` and its documentation from 
`ChapelIteratorSupport.chpl` to a new automatically included standard
module, `VectorizingIterator.chpl`. I am expecting that the functions in
this module will be deprecated once issue #16404 is resolved.

- [x] full local testing

Reviewed by @ronawho - thanks!

### CODE REVIEW

**Comment by bradcray** [comment]
Did we fail to agree to make this module not be auto-used, or is the thought that if it's going away soon anyway, it's easier for everyone using it (hopefully nobody?) to just keep it being auto-used until then?

**Comment by mppf** [comment]
I don't think we've tried to agree about whether or not it is auto-used but I'm expecting that we'll just deprecate these functions soon in favor of `foreach`. So I think it's moot (unless something unexpected happens with the `foreach` direction).

File: modules/internal/fixInternalDocs.sh [hunk 1]
```diff
@@ -131,17 +131,6 @@ removeUsage $file
 ## End ChapelIO ##
 
 
-## ChapelIteratorSupport ##
-
-file="./ChapelIteratorSupport.rst"
-removePrefixFunctions $file
-fixTitle "Vectorizing Iterator" $file
-removeUsage $file
```
Comment (ronawho) [inline modules/internal/fixInternalDocs.sh:139]: The current documentation renders as "to get this `use VectorizingIterator`, but note that it's auto-used so you don't need to `use` it", which seems a little odd

Comment (ronawho) [reply modules/internal/fixInternalDocs.sh:139]: e.g.:
Screen Shot 2020-12-14 at 10 25 28 AM

Comment (mppf) [reply modules/internal/fixInternalDocs.sh:139]: Right, we have other modules that look the same way, e.g. https://chapel-lang.org/docs/master/modules/standard/Math.html

I think it's reasonable to fix that but I wasn't viewing it as a requirement for this PR specifically.

File: compiler/passes/scopeResolve.cpp [hunk 2]
```diff
@@ -2797,6 +2797,7 @@ static bool symbolInBuiltinModule(Symbol* sym) {
   ModuleSymbol* mod = sym->getModule();
   if (mod->modTag == MOD_STANDARD &&
       (!strcmp(mod->name, "Builtins") ||
+       !strcmp(mod->name, "VectorizingIterator") ||
```
Comment (ronawho) [inline compiler/passes/scopeResolve.cpp:2800]: I haven't looked at what `symbolInBuiltinModule` is used for, but it seems unfortunate to have to maintain this list in the compiler (and seems easy to forget to update if you're making a module builtin). Would this be better as a pragma on the module or something?

Separately maybe this list, however we determine it, could be used to squash chpldocs "Here's how to use this module" info?

**Comment by ronawho** [comment]
I was also going to suggest we make this not included by default, but if we expect it to be deprecated soon I agree it makes it easier to have it included by default.

File: compiler/passes/scopeResolve.cpp (hunk 2 above)
Comment (mppf) [reply compiler/passes/scopeResolve.cpp:2800]: Yep. I think if we move it to a pragma on the module we can get chpldoc to squash the unnecessary part. I would like to make these changes in a separate PR (to merge after this one and the ones moving ChapelIO and ChapelEnv).

Comment (mppf) [reply compiler/passes/scopeResolve.cpp:2800]: BTW it is used for the functionality that lists available symbols.

Comment (mppf) [reply compiler/passes/scopeResolve.cpp:2800]: e.g. `trivial/diten/printvisible`

File: modules/internal/fixInternalDocs.sh (hunk 1 above)
Comment (ronawho) [reply modules/internal/fixInternalDocs.sh:139]: Gotcha -- could you create an issue to capture that?

**Review by ronawho — APPROVED** [review]

File: modules/internal/fixInternalDocs.sh (hunk 1 above)
Comment (mppf) [reply modules/internal/fixInternalDocs.sh:139]: PR #16861 will do it.

File: compiler/passes/scopeResolve.cpp (hunk 2 above)
Comment (mppf) [reply compiler/passes/scopeResolve.cpp:2800]: PR #16861 will do it.

**Pull request merged**